### DISCUSSION TITLE
"Show message details" option

### DISCUSSION THREAD

**Discussion by guillezcurra** [tomtau/superbasic-im#12] [Ideas]
Sometimes when I am reading a group chat I miss knowing if people has reacted to a certain message. Maybe a message can be displayed to indicate that it has reactions? Or, instead of this, there could be a dedicated page for detailed information about a message (reactions, seen, sent...). 
I know this could distract people from the important thing, communication. What do you think?

Thank you!

**Comment by tomtau** [comment]
Yes, I think at least delivered/read notification could be useful. Probably OK in 1-to-1 chats, but could be quite distracting or too long in group chats, so perhaps a dedicated page or a hidden expandable section for each message (I'm not sure which HTML/CSS/JS option to do it would work most reliably across feature phone web browsers)?

**Reply by guillezcurra** [reply]
I don't have enough knowledge to know if feature phone web browsers accept something other than static web pages, maybe if the information is already loaded in the html page, but "hidden" it could work for everyone?
Or we could create a dedicated page with a "back to chat" button in it that we know for sure works.

**Reply by tomtau** [reply]
Do you have Opera Mini? Maybe if it can support it, it should be fine as a baseline?

**Reply by guillezcurra** [reply]
I have Opera Mini but it doesn't work well, the native symbian browser works really good and that's what I use.

**Reply by tomtau** [reply]
I guess we can test with both. If the feature works on both, especially Opera Mini, given how poor Opera Mini's browser feature support is, then it should be good to go?

**Reply by guillezcurra** [reply]
I will try to implement it, but I am not sure if I will be able to open the chat in Opera Mini, can you also test it?

**Reply by tomtau** [reply]
There was also this emulator https://web.archive.org/web/20120224031718/http://www.microemu.org/ that could run the J2ME version of Opera Mini:

http://m.opera.com/?act=opts&rnd=1976603211&vid=0x9c7f9c859a2dd90a&ua=All

**Reply by tomtau** [reply]
[microemulator-2.0.4.zip](https://github.com/user-attachments/files/17391050/microemulator-2.0.4.zip)
[opera-mini-8.0.35626-advanced-en.jar.zip](https://github.com/user-attachments/files/17391064/opera-mini-8.0.35626-advanced-en.jar.zip)

**Reply by guillezcurra** [reply]
Thanks, I have been busy but I will try to develop this and try it. I have researched and I have found an html element called details that allows this behavior without the need of javascript. This is what I will try first and test it everywhere I can.
https://developer.mozilla.org/en-US/docs/Web/HTML/Element/details

**Reply by tomtau** [reply]
Looks good, but not sure about Opera Mini ("This feature is well established and works across many devices and browser versions. It’s been available across browsers since January 2020.")... for Opera Mini, it's more like features that existed in 2005 (if even that) 😂

**Reply by guillezcurra** [reply]
I was trying to create a message details page and it seems impossible to retrieve a message by Id in whatsapp-web.js.
https://github.com/pedroslopez/whatsapp-web.js/issues/1721
Its weird that this is not possible o_0, I will try this drop down details option instead.

**Reply by guillezcurra** [reply]
I have implemented these lines of code:
```html
<details>
        <summary><small>Details</small></summary>
        {{#if this.hasReaction}}
            <small>This message has reactions</small></br />
        {{/if}}
        {{#if this.hasQuotedMessage}}
            <small>This message is a reply to another message</small></br />
        {{/if}}
</details>
```

But, as you said, it is too new for my browser.

**Reply by tomtau** [reply]
Maybe it could just be divs with some simple CSS style and a custom JS code to either add/remove div nodes or add visibility: hidden? Opera Mini can handle some simple JS with some limitations (the scripts needs to be initiated by users such as by clicking a button, not running more than 5 seconds etc.).

Or the option whether to display reactions in a message can be global or per chat room (one can see the chatroom with all reactions expanded or without any reactions)?

**Reply by guillezcurra** [reply]
I have managed to make it work for Opera Mini 4.4, I will add useful info to the details section and then I will create a merge request. Thank you!
This is the approach I have finally used:
```html
<small>
          <a href="#" id="pitjar detalls" onclick="document.getElementById('details{{this.id}}').classList.toggle('hidden'); return false;">Details</a>
          <div id="details{{this.id}}" class="hidden">
                  {{#if this.hasReaction}}
                  <small>This message has reactions</small><br />
                  {{/if}}
                  {{#if this.hasQuotedMessage}}
                  <small>This message is a reply to another message</small><br />
                  {{/if}}
          </div>
</small>
 ```

**Comment by guillezcurra** [comment]
I have tried using javascript but doesn't work, works fine from my iphone.
I have tried a css+html approach but also doesn't work in my Nokia e52.
```html
<style>
  .details-content {
    display: none;
  }
  input[type="checkbox"] {
    display: none;
  }
  input[type="checkbox"] + label {
    display: inline-block;
    padding: 5px;
    background-color: #FFFFFF;
    color: black;
    border: none;
    cursor: pointer;
    text-align: center;
  }
  input[type="checkbox"]:checked + label {
    background-color: #0056b3;
  }
  input[type="checkbox"]:checked + label + .details-content {
    display: block;
  }
</style>
.....
<div class="details">
     <input type="checkbox" id="toggle">
     <label for="toggle">Details</label>
     <div class="details-content">
                {{#if this.hasReaction}}
                  <small>This message has reactions</small><br />
                {{/if}}
                {{#if this.hasQuotedMessage}}
                  <small>This message is a reply to another message</small><br />
                {{/if}}
     </div>
</div>
```

Maybe it is not possible
